packages/@aws-cdk/aws-stepfunctions/lib/states/custom-state.ts

```diff
@@ -0,0 +1,53 @@
+import * as cdk from '@aws-cdk/core';
+import { Chain } from '..';
+import { IChainable, INextable } from '../types';
+import { State } from './state';
+
+/**
+ * Properties for defining a custom state definition
+ */
+export interface CustomStateProps {
+  /**
+   * Amazon States Language (JSON-based) definition of the state
+   *
+   * @see https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html
+   */
+  readonly stateJson: { [key: string]: any };
+}
+
+/**
+ * Custom state in the state machine
+ *
+ */
+export class CustomState extends State implements IChainable, INextable {
+  public readonly endStates: INextable[];
+
+  /**
+   * Amazon States Language (JSON-based) definition of the state
+   *
+   * @see https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html
+   */
+  public readonly stateJson: { [key: string]: any};
+
+  constructor(scope: cdk.Construct, id: string, props: CustomStateProps) {
+    super(scope, id, {});
+
+    this.endStates = [this];
+    this.stateJson = props.stateJson;
+  }
+
+  /**
+   * Continue normal execution with the given state
+   */
+  public next(next: IChainable): Chain {
+    super.makeNext(next.startState);
+    return Chain.sequence(this, next);
+  }
+
+  /**
+   * Returns the Amazon States Language object for this state
+   */
+  public toStateJson(): object {
+    return this.stateJson;
```

Comment:
Since this thing is a nextable, probably render the `Next` into here as well (if it is set).